packages/nimble-components/src/patterns/button/tests/states.ts

```diff
@@ -0,0 +1,23 @@
+import { pascalCase } from '@microsoft/fast-web-utilities';
+import { ButtonAppearance, ButtonAppearanceVariant } from '../types';
+
+/* array of iconVisible, labelVisible, endIconVisible */
+export const partVisibilityStates = [
+    [true, true, false],
+    [true, false, false],
+    [false, true, false],
+    [true, true, true],
+    [false, true, true]
+] as const;
+export type PartVisibilityState = (typeof partVisibilityStates)[number];
+
+export const appearanceStates: [string, string | undefined][] = Object.entries(
+    ButtonAppearance
+).map(([key, value]) => [pascalCase(key), value]);
+export type AppearanceState = (typeof appearanceStates)[number];
+
+export const appearanceVariantStates: [string, string | undefined][] = Object.entries(ButtonAppearanceVariant).map(([key, value]) => [
+    pascalCase(key),
+    value
+]);
```

Comment:
You lose `as const`-ness typing of the enum states this way. Also it's not easier to read and not really changing often so don't think being explicit is a maintenance burden. Think it's better to have the stronger type

![image](https://github.com/ni/nimble/assets/1588923/cdb71f24-09f3-4785-8130-1849a6e5e05d)

Compared to

![image](https://github.com/ni/nimble/assets/1588923/f47223f2-d37b-44c1-aab7-27a231031f9b)

Reply:
I had just copied these as they were. I'll change them.